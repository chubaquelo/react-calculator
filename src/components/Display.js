import React from 'react';
import PropTypes from 'prop-types';

function Display(props) {
  const { total, operation, next } = props;
  return (
    <div className="bg-black text-white text-5xl w-full h-18 p-2 flex justify-end items-end rounded-t-lg border-white border-4 border-b-0 select-all">
      {`${total} ${operation} ${next}`}
    </div>
  );
}

Display.propTypes = {
  total: PropTypes.string,
  next: PropTypes.string,
  operation: PropTypes.string,
};

Display.defaultProps = {
  total: '0',
  next: PropTypes.string,
  operation: PropTypes.string,
};

export default Display;
