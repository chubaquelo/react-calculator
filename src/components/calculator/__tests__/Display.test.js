import React from 'react';
import renderer from 'react-test-renderer';
import Display from '../Display';

describe.only('Test Display Snapshot', () => {
  it('matches Display snapshot', () => {
    const forTest = renderer.create(<Display />).toJSON();
    expect(forTest).toMatchSnapshot();
  });
});
