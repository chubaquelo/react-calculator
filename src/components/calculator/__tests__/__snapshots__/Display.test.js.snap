// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Display Snapshot matches Display snapshot 1`] = `
<div
  className="break-all text-right bg-black text-white text-5xl w-full h-18 p-2 flex justify-end items-end rounded-t-lg border-white border-4 border-b-0 select-all"
>
  0function () { [native code] }function () { [native code] }
</div>
`;
